src/main/java/me/desht/pneumaticcraft/common/thirdparty/cofhcore/HoldingEnchantableProvider.java

```diff
@@ -17,66 +17,41 @@
 import me.desht.pneumaticcraft.lib.ModIds;
-import net.minecraft.core.Direction;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.TextComponent;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.enchantment.Enchantment;
 import net.minecraft.world.item.enchantment.EnchantmentHelper;
-import net.minecraftforge.common.capabilities.Capability;
-import net.minecraftforge.common.capabilities.CapabilityManager;
-import net.minecraftforge.common.capabilities.CapabilityToken;
-import net.minecraftforge.common.capabilities.ICapabilityProvider;
-import net.minecraftforge.common.util.LazyOptional;
 import net.minecraftforge.registries.ForgeRegistries;
 
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
 import java.util.List;
 
-public class HoldingEnchantableProvider implements ICapabilityProvider {
-    public static final Capability<IEnchantableItem> CAPABILITY_ENCHANTABLE_ITEM = CapabilityManager.get(new CapabilityToken<>() {});
+public class HoldingEnchantableProvider {
     static Enchantment holdingEnchantment = null;
-
-    private final AllowHoldingEnchant ench = new AllowHoldingEnchant();
-    private final LazyOptional<IEnchantableItem> lazy = LazyOptional.of(() -> ench);
-
-    @Nonnull
-    @Override
-    public <T> LazyOptional<T> getCapability(@Nonnull Capability<T> cap, @Nullable Direction side) {
-        return CAPABILITY_ENCHANTABLE_ITEM.orEmpty(cap, lazy);
-    }
+    private static final boolean holdingEnabled = ConfigHelper.common().integration.cofhHoldingMultiplier.get() > 0;
 
     static void registerVolumeModifier() {
+        // Gets if Holding enchantment has been registered
         holdingEnchantment = ForgeRegistries.ENCHANTMENTS.getValue(new ResourceLocation(ModIds.COFH_CORE, "holding"));
-        if (holdingEnchantment != null) {
-            PneumaticRegistry.getInstance().getItemRegistry().registerPneumaticVolumeModifier(new COFHVolumeModifier(holdingEnchantment));
-        }
-    }
-
-    static void registerEnchantment() {
-        CoreEnchantments.registerHoldingEnchantment();
-    }
 
-    public static class AllowHoldingEnchant implements IEnchantableItem {
-        @Override
-        public boolean supportsEnchantment(Enchantment enchantment) {
-            return ConfigHelper.common().integration.cofhHoldingMultiplier.get() > 0 && enchantment == holdingEnchantment;
+        // Registers the volume modifier for the Holding enchant if it's present and config-enabled
+        if (holdingEnchantment != null && holdingEnabled) {
+            PneumaticRegistry.getInstance().getItemRegistry().registerPneumaticVolumeModifier(new C0FHVolumeModifier(holdingEnchantment));
         }
     }
 
-    public record COFHVolumeModifier(Enchantment holding) implements ItemVolumeModifier {
+    public record C0FHVolumeModifier(Enchantment holding) implements ItemVolumeModifier {
```

Comment:
Any particular reason for the 0 (zero) instead of O (capital 'o') here?

Reply:
I honestly have no idea why I changed that, must have been a typo when I was retyping the name or something.